cmd/create.go

```diff
@@ -86,7 +86,20 @@ A full list of component types that can be deployed is available using: 'odo cat
 			os.Exit(1)
 		}
 
-		componentImageName, componentType, componentName, componentVersion := util.ParseCreateCmdArgs(args)
+		componentImageName, componentType, defaultComponentName, componentVersion := util.ParseCreateCmdArgs(args)
+
+		componentList, err := component.List(client, applicationName, projectName)
+		checkError(err, "")
+
+		// Get component names
+		var componentNames []string
+		for _, componentInfo := range componentList {
+			componentNames = append(componentNames, componentInfo.Name)
+		}
+
+		// Generate unique name for component
+		componentName, err := component.GetDefaultComponentName(defaultComponentName, componentNames)
```

Comment:
Again, see other comment, I believe that all of this can be shortened into only 4 lines (passing in `apps` or `componentList` into a funciton that checks to see if the new app name exists..)